recipes-support/boost/files/0001-add-typedef-executor_type.patch

```diff
@@ -0,0 +1,56 @@
+From 99ad3ebac0b7466d57c98fb9583fbe8789729691 Mon Sep 17 00:00:00 2001
+From: Patrick Vacek <patrickvacek@gmail.com>
+Date: Wed, 4 Mar 2020 12:41:02 +0000
+Subject: [PATCH] add typedef executor_type
+
+Backported from here:
+https://github.com/boostorg/process/commit/6a4d2ff72114ef47c7afaf92e1042aca3dfa41b0
+
+Suggested-by: Kostiantyn Bushko <kbushko@intellias.com>
+Signed-off-by: Patrick Vacek <patrickvacek@gmail.com>
+---
+ boost/process/async_pipe.hpp                | 2 ++
+ boost/process/detail/posix/async_pipe.hpp   | 1 +
+ boost/process/detail/windows/async_pipe.hpp | 1 +
+ 3 files changed, 4 insertions(+)
+
+diff --git a/boost/process/async_pipe.hpp b/boost/process/async_pipe.hpp
+index 101fe1d..a562432 100644
+--- a/boost/process/async_pipe.hpp
++++ b/boost/process/async_pipe.hpp
+@@ -47,6 +47,8 @@ public:
+      */
+     typedef platform_specific handle_type;
+ 
++    typedef typename handle_type::executor_type executor_type;
++
+     /** Construct a new async_pipe, does automatically open the pipe.
+      * Initializes source and sink with the same io_context.
+      * @note Windows creates a named pipe here, where the name is automatically generated.
+diff --git a/boost/process/detail/posix/async_pipe.hpp b/boost/process/detail/posix/async_pipe.hpp
+index 725a078..a82c057 100644
+--- a/boost/process/detail/posix/async_pipe.hpp
++++ b/boost/process/detail/posix/async_pipe.hpp
+@@ -23,6 +23,7 @@ class async_pipe
+ public:
+     typedef int native_handle_type;
+     typedef ::boost::asio::posix::stream_descriptor handle_type;
++    typedef typename handle_type::executor_type executor_type;
+ 
+     inline async_pipe(boost::asio::io_context & ios) : async_pipe(ios, ios) {}
+ 
+diff --git a/boost/process/detail/windows/async_pipe.hpp b/boost/process/detail/windows/async_pipe.hpp
+index 06d5f2d..0b447f9 100644
+--- a/boost/process/detail/windows/async_pipe.hpp
++++ b/boost/process/detail/windows/async_pipe.hpp
+@@ -48,6 +48,7 @@ class async_pipe
+ public:
+     typedef ::boost::winapi::HANDLE_ native_handle_type;
+     typedef ::boost::asio::windows::stream_handle   handle_type;
++    typedef typename handle_type::executor_type executor_type;
+ 
+     async_pipe(boost::asio::io_context & ios) : async_pipe(ios, ios, make_pipe_name(), true) {}
+     async_pipe(boost::asio::io_context & ios_source, boost::asio::io_context & ios_sink)
+-- 
```

Comment:
line: 54-56

Reply:
Where is the patch that you're using? I'm struggling to find it.

FWIW I made this patch with `git format-patch` after manually adding the lines into the source tree of boost built by bitbake.

Reply:
Maybe it's not important at all, I just have the diff locally as it required only for the building of the tests on the mac and we don't need it for the building of just aktualizr.

`mike@msul:~/work/here/projects/process$ git diff e6f478e 0a554c9
diff --git a/include/boost/process/async_pipe.hpp b/include/boost/process/async_pipe.hpp
index a562432..101fe1d 100644
--- a/include/boost/process/async_pipe.hpp
+++ b/include/boost/process/async_pipe.hpp
@@ -47,8 +47,6 @@ public:
      */
     typedef platform_specific handle_type;
 
-    typedef typename handle_type::executor_type executor_type;
-
     /** Construct a new async_pipe, does automatically open the pipe.
      * Initializes source and sink with the same io_context.
      * @note Windows creates a named pipe here, where the name is automatically generated.
diff --git a/include/boost/process/detail/posix/async_pipe.hpp b/include/boost/process/detail/posix/async_pipe.hpp
index a82c057..725a078 100644
--- a/include/boost/process/detail/posix/async_pipe.hpp
+++ b/include/boost/process/detail/posix/async_pipe.hpp
@@ -23,7 +23,6 @@ class async_pipe
 public:
     typedef int native_handle_type;
     typedef ::boost::asio::posix::stream_descriptor handle_type;
-    typedef typename handle_type::executor_type executor_type;
 
     inline async_pipe(boost::asio::io_context & ios) : async_pipe(ios, ios) {}
 
diff --git a/include/boost/process/detail/windows/async_pipe.hpp b/include/boost/process/detail/windows/async_pipe.hpp
index 0b447f9..06d5f2d 100644
--- a/include/boost/process/detail/windows/async_pipe.hpp
+++ b/include/boost/process/detail/windows/async_pipe.hpp
@@ -48,7 +48,6 @@ class async_pipe
 public:
     typedef ::boost::winapi::HANDLE_ native_handle_type;
     typedef ::boost::asio::windows::stream_handle   handle_type;
-    typedef typename handle_type::executor_type executor_type;
 
     async_pipe(boost::asio::io_context & ios) : async_pipe(ios, ios, make_pipe_name(), true) {}
     async_pipe(boost::asio::io_context & ios_source, boost::asio::io_context & ios_sink)
`

Reply:
Looks the same to me; what am I missing?